### PR TITLE
feat: DAH-1995 Update Lottery Pref Section on Listing Details Page

### PR DESCRIPTION
**Pull request by jimlin-sfgov**
https://sfgovdt.jira.com/browse/DAH-1995

### CODE REVIEW

**Review by cade-exygy — APPROVED** [review]

**Comment by alulabeshue-sfgov** [comment]
Can we make this PR to the [release branch](https://github.com/SFDigitalServices/sf-dahlia-web/pull/2051). There are some new changes in main that I think we can avoid adding to prod deployment this late.

File: app/javascript/modules/listingDetails/ListingDetailsEligibility.tsx [hunk 1]
```diff
@@ -287,9 +288,29 @@ export const ListingDetailsEligibility = ({
...
+                      <b>{t("listingsForSale.lotteryPreferences.priorityForUsMilitaryVeterans")}</b>
+                    </div>
+                    <div className="mb-4">
+                      {t("listingsForSale.lotteryPreferences.veteransGetPriority")}
+                    </div>
+                    <div>
+                      <Link
+                        className="text-blue-700"
+                        external={true}
+                        href="https://www.sf.gov/get-priority-housing-lottery-if-you-are-veteran"
```
[the remark this answers is not in the document]
Comment (jimlin-sfgov) [reply app/javascript/modules/listingDetails/ListingDetailsEligibility.tsx:306]: Good call, the new ticket would probably update this function to be compatible with "non-node" sf gov urls:
https://github.com/SFDigitalServices/sf-dahlia-web/blob/fbde74e11f639b43d2560a7abd6d0e0a5f51d683/app/javascript/util/languageUtil.tsx#L148

**Comment by jimlin-sfgov** [comment]
> Is this an intended change in the ticket?

Yeah, I actually missed that part of the requirements. Thanks for catching

**Review by alulabeshue-sfgov — COMMENTED** [review]
**question:** If in the future we will change these to human translations do we need a follow up ticket or do you plan to handle as part of this ticket?

File: app/javascript/modules/listings/SharedHelpers.tsx [hunk 2]
```diff
@@ -95,3 +98,9 @@ export const getEventNote = (listingEvent: ListingEvent) => {
     </div>
   )
 }
+
+export const listingHasVeteransPreference = (listing: RailsListing): boolean => {
+  return !!listing.Listing_Lottery_Preferences?.some((preference: ListingLotteryPreference) =>
+    preference?.Lottery_Preference?.Name?.toLowerCase().includes("veteran")
+  )
+}
```
[the remark this answers is not in the document]
Comment (jimlin-sfgov) [reply app/javascript/modules/listings/SharedHelpers.tsx]: @alulabeshue-sfgov changes have been made per your suggestions

**Comment by jimlin-sfgov** [comment]
> Can we make this PR to the https://github.com/SFDigitalServices/sf-dahlia-web/pull/2051?

@alulabeshue-sfgov Done, note that there are now changes in the diff that were not originally part of the PR.

**Comment by jimlin-sfgov** [comment]
per discussion with @alulabeshue-sfgov I'm switching the base branch back to main, because there's no way around _not_ deploying certain changes now.

**Review by alulabeshue-sfgov — COMMENTED** [review]
LGTM, thanks @jimlin-sfgov for the changes!

**Pull request merged**